api_app/analyzers_manager/observable_analyzers/greynoise_labs.py

```diff
@@ -0,0 +1,144 @@
+        "ip_required": True,
+    },
+    "topknocks": {
+        "query_string": "query TopKnocks($ip: String) { topKnocks(ip: $ip) \
+            { queryInfo { resultsAvailable resultsLimit } knock { last_crawled \
+            last_seen source_ip knock_port title favicon_mmh3_32 \
+            favicon_mmh3_128 jarm ips emails links tor_exit headers apps } } } ",
+        "ip_required": True,
+    },
+    "topc2s": {
+        "query_string": "query TopC2s { topC2s { queryInfo \
+            { resultsAvailable resultsLimit } c2s { source_ip c2_ips \
+            c2_domains payload hits pervasiveness } } } ",
+        "ip_required": False,
+        "db_location": db_location,
+    },
+}
+
+
+class GreynoiseLabs(ObservableAnalyzer):
+    _auth_token: str
+
+    def run(self):
+        result = {}
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self._auth_token}",
+        }
+
+        try:
+            for key, value in queries.items():
+                if not value["ip_required"]:
+                    if not os.path.isfile(value["db_location"]) and not self.update():
+                        raise AnalyzerRunException(f"Failed extraction from {key} db")
+                    if not os.path.exists(value["db_location"]):
+                        raise AnalyzerRunException(
+                            f"database location {value['db_location']} does not exist"
+                        )
+
+                    with open(value["db_location"], "r", encoding="utf-8") as f:
+                        db = f.read()
+
+                    db_list = db.split("\n")
+                    if self.observable_name in db_list:
+                        result[key] = {"found": True}
+                    else:
+                        result[key] = {"found": False}
+                    continue
+
+                json_body = {
+                    "query": value["query_string"],
+                    "variables": {"ip": f"{self.observable_name}"},
+                }
+                response = requests.post(headers=headers, json=json_body, url=url)
+                response.raise_for_status()
+                result[key] = response.json()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        return result
+
+    @classmethod
+    def _get_auth_token(cls):
+        for plugin in PluginConfig.objects.filter(
+            parameter__python_module=cls.python_module,
+            parameter__is_secret=True,
+            parameter__name="auth_token",
+        ):
+            if plugin.value:
+                return plugin.value
+        return None
+
+    @classmethod
+    def _update_db(cls, auth_token: str):
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {auth_token}",
+        }
+
+        try:
+            logger.info("Fetching data from greynoise API.....")
```

Comment:
please log the  observable name too here otherwise static logs like this are useless considering that the base class already generates some generic logs like this